### DISCUSSION TITLE
How can we improve the docs for you?

### DISCUSSION THREAD

**Discussion by melohagan** [Budibase/budibase#9488] [Polls]
Hey there Buddies!
 
I am looking to improve the documentation to try and help users get more out of Budibase and get started easier. 
There's a number of things I have in mind, but is there anything you would like to see prioritised? 

I would also appreciate any feedback on how often you use https://docs.budibase.com/docs and what would make them better for you.

 - How do you find the documentation compared with community support (Github Discussions and Discord) - do you have a preference? 
 - Have you watched any of the [Budibase University](https://vimeo.com/showcase/budibase-university), and if so did you prefer the video tutorials over text, or like the combination of both?

As well as dropping a comment, please take a moment to answer the poll question below. Really appreciate it!
If there's any missing option or other feedback please do share.

**Comment by renrom** [comment]
Hi,

I just started with Budibase and i have read the Docs on the website several times. They are helpfull for a starter and the video's are a very helpfull addition! the video's shows more often the tricks i needed to get started or to get further.

What i missed, was for example the using of the {URL} bindings and more clear examples how to pass data from one screen to another. 

Also the use of Handlebars is not always very clear for me and how to use them in the different kind of screens, i think the misunderstanding is mostly because of the autosearch which doesn't complete the full syntax i gues, mostly missing the field names.

Overall, i think this project is one of a kind, i did check out others, for example appsmith, but personnaly i think budibase has more potential. 

Good work! and i will struggle further with budibase! :)

**Reply by melohagan** [reply]
Great feedback - thank you! 

There are some ideas for making the bindings a bit easier to work with, and is in the roadmap for roughly Q2: https://github.com/Budibase/budibase/projects/26

**Reply by renrom** [reply]
Sounds good!

What also would help is a (complete) sample application, based on an external datasource, for example MySQL with examples how to setup releationship with some (simple?) SQL.  in the documents there is some small examples to Pets -- Owners -- Appointments and also studens, teachers, parkings. But a downloaded example to learn from the design would be great.

**Reply by melohagan** [reply]
I do have a MySQL "start-to-finish" tutorial mostly written up. There's actually a couple of minor bug fixes to go out with the upcoming release and then I'll be able to publish that.

**Reply by renrom** [reply]
Great!

**Comment by juangbm** [comment]
Hi,
I think the current documentation is good enough to start with a simple app, but I think the problems came when you want to do something more complex, I think documentation about data management and how to treat info from different types (relationships, data provider, option pickers, ...)  will help a lot.
Thanks!

**Comment by Bitello** [comment]
Hi @melohagan

This initiative is commendable!
All my comments are intended to be constructive.

Some time ago you helped me to solve a problem using the "Update State".
Something that happens is that I am in doubt regarding some important feature of the tool (such as the "update state"), I do a search and I don't find a reference. If I can't find it in terms of "what I see", what about resources that are available but not documented.

Another point that causes me a bit of confusion is the focus of the documentation on BudiDB. The examples of the tool are given on the native database, but it has (since it is still in evolution) some "limitations".
I wonder if the focus of the documentation shouldn't be on a "stable database" and more "feature-rich" (while BudiDB doesn't reach the state of the art)...

**Reply by melohagan** [reply]
Thanks for the feedback! 

That is a good point about making Budibase concepts such as **Update State** be more accessible to new users who aren't aware of the jargon!

The focus on the BudibaseDB in part is because even when using a SQL database the data-types are consistent.
For example if you have a SQL table with a *varchar* column and a *smallint* column, Budibase will interpret these as *Text* and *Number* respectively. 
There are some ideas within the company for improving upon some of the limitations of the internal Budibase tables.

**Comment by christophsanz** [comment]
The documentation for the automations is a bit confusing. I first thought a "loop" would not only loop this one step, but also all other steps after. It took quite a bit of time till i figured out how to access the outputs of the previous loops and loop over them again in the next step - if that makes sense 😅
I also really hope that the automations get overhauled, adding or updating a few hundred rows at once is unbearably slow because every row is added one after another (or i am doing something wrong ).

**Reply by melohagan** [reply]
Thanks for the feedback! 

Automations can be fairly complex, so some more examples, particularly around the looping, would be good.

You are correct in that there currently is no batch update/insert/delete, and that would be a performance improvement as well.

**Comment by Simoon-F** [comment]
I see that the js loaded by the official website hosting platform is CDN, .... I didn't find the configuration method of CDN in the document, which may be the reason why I am not familiar with it. Is it possible to add this point in the document?

**Reply by Simoon-F** [reply]
> I see that the js loaded by the official website hosting platform is CDN, .... I didn't find the configuration method of CDN in the document, which may be the reason why I am not familiar with it. Is it possible to add this point in the document?

Okay, I 've worked it out. Indeed, I am not skilled enough, and the official website documents need to be improved and not clear enough. I found out what my problem was from the source code.

**Comment by kzkondo** [comment]
Hi! @melohagan Thank you for your prompt response!
I love budibase.
I am using it in Japan.
I've been using it as a self-host since November 2022, and since I can't translate English, I rely on Google Translate to enjoy development.
Sorry for the poor English.

The binding documentation is interesting.
Especially handlebars, helpers.
There are many functions, but most of them are not used,
It is very easy to understand when concrete examples are introduced in the video.

**Comment by AlejandroFNadal** [comment]
I would love more info on return types. Example: I use a Datepicker, and I save the value of it to a element of an object in a state. What is the type there? Is it a date? is it a string? I have discovered all this stuff by trial and error, but it was a bit cumbersome to do.

**Comment by helena-pa** [comment]
What I miss the most is the explanation of complete use cases, from start to finish, with component interactions. Especially with external data, because internal or SQL data is much better documented.

Having a complete test case with external data would also be useful since in this case it is not possible to auto generate screens. Maybe it could be a template?

And for me the most practical thing is clearly to have a good written documentation. Videos are a bonus.

**Reply by melohagan** [reply]
Hey Helena, thanks very much for the feedback!

When it comes to external data - do you have a particular datasource(s) in mind?

I did try to include more complete use-case examples for the SQL datasources, e.g. [MS SQL Server](https://docs.budibase.com/docs/ms-sql-server)

There are a number of videos in our [Budibase university](https://vimeo.com/showcase/budibase-university). Had you watched any of them, and did you find them useful?

**Reply by helena-pa** [reply]
I mean external non-SQL data sources (I'm using Airtable right now).

I saw most of the videos in Budibase university. It's interesting to get familiar with the interface and some concepts, but as soon as you go further, you are quickly limited.

**Comment by strattoncharles** [comment]
Hi,

I am very interested, but I would love to see instructions for installing and hosting on Microsoft windows server 2019 and 2022 in IIS. In iis we can host wordpress etc. I would love to be able to host this on windows servers.

**Comment by sottey** [comment]
I know this is SUPER old, and I am late to the party, but I sure would love to have either github access to the doc source OR have the docs in PDF format. This would allow people to train an AI to be their Budibase expert without having to hammer your servers.

Thanks!
Sean
